frontend/src/components/c-ramp.vue

```diff
@@ -111,29 +121,31 @@ export default {
       if (this.isDeletesContribution(slice)) {
         return this.deletesContributionRampSize;
       }
-      const newSize = 100 * (slice.insertions / this.avgsize);
+      // '+' unary operator here attempts to convert this.avgsize to number, if it is not already one
+      const newSize = 100 * (slice.insertions / +this.avgsize);
       return Math.max(newSize * this.rampSize, 0.5);
     },
-    getContributionMessage(slice, commit) {
-      let title = '';
-      if (this.tframe === 'commit') {
+    // commit of type CommitResult must be provided if tframe === 'commit', only pass null if tframe !== 'commit
+    getContributionMessage(slice: Commit, commit: CommitResult | null) {
+      if (this.tframe === 'commit' && commit !== null) {
```

Comment:
Instead of enforcing that the caller must pass null as the second argument when `tframe` is not `commit`, which might not be very obvious and could potentially be missed out, do you think it would be better to separate the two scenarios into different methods instead?

Reply:
Probably a great idea to do so, as this may cause confusion. Will update to incorporate your suggestion.